docs/source/tutorials/arch_modeling/Spypads_tutorial.rst

```diff
@@ -0,0 +1,334 @@
+
+.. code-block:: bash
+
+   python3 openfpga_flow/scripts/run_fpga_task.py fpga_verilog/spypad --debug --show_thread_logs
+
+.. note:: Python 3.8 or later is required to run this task
+
+We can now see the instantiation of these spypads in ``fpga_top.v`` and ``luts.v``. We will start by viewing ``luts.v`` with the following command:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/tasks/fpga_verilog/spypad/latest/k6_frac_N10_tileable_adder_register_scan_chain_depop50_spypad_40nm/and2/MIN_ROUTE_CHAN_WIDTH/SRC/sub_module/luts.verilog
+
+The spypads are coming from the ``frac_lut6_spypad`` circuit model. In ``luts.v``, the ``frac_lut6_spypad`` module is defined around **LINE150** and looks as follows:
+
+.. code-block:: verilog
+
+   module frac_lut6_spypad(in,
+   sram,
+   sram_inv,
+   mode,
+   mode_inv,
+   lut4_out,
+   lut5_out,
+   lut6_out);
+   //----- INPUT PORTS -----
+   input [0:5] in;
+   //----- INPUT PORTS -----
+   input [0:63] sram;
+   //----- INPUT PORTS -----
+   input [0:63] sram_inv;
+   //----- INPUT PORTS -----
+   input [0:1] mode;
+   //----- INPUT PORTS -----
+   input [0:1] mode_inv;
+   //----- OUTPUT PORTS -----
+   output [0:3] lut4_out;
+   //----- OUTPUT PORTS -----
+   output [0:1] lut5_out;
+   //----- OUTPUT PORTS -----
+   output [0:0] lut6_out;
+
+The ``fpga_top.v`` file has some similarities. We can view the ``fpga_top.v`` file by running the following command:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/tasks/fpga_verilog/spypad/latest/k6_frac_N10_tileable_adder_register_scan_chain_depop50_spypad_40nm/and2/MIN_ROUTE_CHAN_WIDTH/SRC/fpga_top.v
+
+If we look at the module definition and ports of ``fpga_top.v`` we should see the following:
+
+.. code-block:: verilog
+
+   module fpga_top(pReset,
+                prog_clk,
+                TESTEN,
+                set,
+                reset,
+                clk,
+                gfpga_pad_frac_lut6_spypad_lut4_out,
+                gfpga_pad_frac_lut6_spypad_lut5_out,
+                gfpga_pad_frac_lut6_spypad_lut6_out,
+                gfpga_pad_GPIO_PAD,
+                ccff_head,
+                ccff_tail);
+   //----- GLOBAL PORTS -----
+   input [0:0] pReset;
+   //----- GLOBAL PORTS -----
+   input [0:0] prog_clk;
+   //----- GLOBAL PORTS -----
+   input [0:0] TESTEN;
+   //----- GLOBAL PORTS -----
+   input [0:0] set;
+   //----- GLOBAL PORTS -----
+   input [0:0] reset;
+   //----- GLOBAL PORTS -----
+   input [0:0] clk;
+   //----- GPOUT PORTS -----
+   output [0:3] gfpga_pad_frac_lut6_spypad_lut4_out;
+   //----- GPOUT PORTS -----
+   output [0:1] gfpga_pad_frac_lut6_spypad_lut5_out;
+   //----- GPOUT PORTS -----
+   output [0:0] gfpga_pad_frac_lut6_spypad_lut6_out;
+   //----- GPIO PORTS -----
+   inout [0:7] gfpga_pad_GPIO_PAD;
+   //----- INPUT PORTS -----
+   input [0:0] ccff_head;
+   //----- OUTPUT PORTS -----
+   output [0:0] ccff_tail;
+
+Using :numref:`fig_gpout` as a guide, we can relate our task like :numref:`fig_gpout_example`

```

Comment:
figure link is broken. Also, please try to remake the figure by considering the architecture you are modeling.